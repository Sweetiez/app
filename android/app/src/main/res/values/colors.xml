<resources>
    <color name="yellow">#dba970</color>
</resources>
