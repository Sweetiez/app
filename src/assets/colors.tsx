const colors = {
  yellow: '#dba970',
  grey: '#9CA3AF',
  lightGrey: '#f1f1f5',
  white: '#ffffff',
  black: '#000000',
  blue: '#4366b2',
  red: '#ff0000',
  green: '#2b7519',
};

export default colors;
