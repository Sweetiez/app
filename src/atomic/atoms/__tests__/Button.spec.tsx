import * as React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';
import colors from '../../../assets/colors';

describe('Button', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button onPress={jest.fn()} text="This is a Button" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with different color', () => {
    const tree = renderer
      .create(
        <Button
          onPress={jest.fn()}
          text="This is a Button"
          color={colors.black}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders loading', () => {
    const tree = renderer
      .create(<Button onPress={jest.fn()} text="This is a Button" isLoading />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders disabled', () => {
    const tree = renderer
      .create(<Button onPress={jest.fn()} text="This is a Button" disabled />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
