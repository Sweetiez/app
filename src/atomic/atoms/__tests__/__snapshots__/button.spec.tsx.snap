// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#dba970",
      "borderRadius": 15,
      "minWidth": 80,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
        },
      ]
    }
  >
    This is a Button
  </Text>
</View>
`;

exports[`Button renders disabled 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9CA3AF",
      "borderRadius": 15,
      "minWidth": 80,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
        },
      ]
    }
  >
    This is a Button
  </Text>
</View>
`;

exports[`Button renders loading 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9CA3AF",
      "borderRadius": 15,
      "minWidth": 80,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ffffff"
  />
</View>
`;

exports[`Button renders with different color 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": 15,
      "minWidth": 80,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
        },
      ]
    }
  >
    This is a Button
  </Text>
</View>
`;
